File: Android/aoe/model-option-loader/src/main/java/com/didi/aoe/library/modeloption/loader/pojos/LocalModelOption.java
package com.didi.aoe.library.modeloption.loader.pojos;

import android.support.annotation.Keep;
import android.text.TextUtils;


import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.didi.aoe.library.api.AoeModelOption;
import com.didi.aoe.library.api.domain.ModelSource;

/**
 * AoE默认的模型配置定义
 *
 * @author noctis
 */
@Keep
public class LocalModelOption implements AoeModelOption {
    private String version;

    /**
     * 模型文件assets目录路径
     */
    private String modelDir;
    /**
     * 模型文件名(带后缀)
     */
    private String modelName;

    @Nullable
    @Override
    public String getVersion() {
        return version;
    }

    @NonNull
    @Override
    public String getModelDir() {
        return modelDir;
    }

    @NonNull
    @Override
    public String getModelName() {
        return modelName;
    }

    @NonNull
    @Override
    public String getModelSource() {
        return ModelSource.LOCAL;
    }

    @Override
    public boolean isValid() {
        return !TextUtils.isEmpty(modelDir) &&
                !TextUtils.isEmpty(modelName);
    }

    @Override
    public String toString() {
        return "LocalModelOption{" +
                "version='" + version + '\'' +
                ", modelDir='" + modelDir + '\'' +
                ", modelName='" + modelName + '\'' +
                '}';
    }
}

File: Android/aoe/core/src/main/java/com/didi/aoe/library/modeloption/loader/pojos/LocalModelOption.kt
package com.didi.aoe.library.modeloption.loader.pojos

import android.support.annotation.Keep
import android.text.TextUtils
import com.didi.aoe.library.api.AoeModelOption
import com.didi.aoe.library.api.domain.ModelSource
import com.google.gson.annotations.SerializedName

/**
 * AoE默认的模型配置定义
 *
 * @author noctis
 */
@Keep
open class LocalModelOption : AoeModelOption {
    @SerializedName("version")
    private val version: String? = null
    /**
     * 模型文件assets目录路径
     */
    @SerializedName("modelDir")
    private var modelDir: String? = null
    /**
     * 模型文件名(带后缀)
     */
    @SerializedName("modelName")
    private var modelName: String? = null

    constructor() {}
    constructor(modelDir: String?, modelName: String?) {
        this.modelDir = modelDir
        this.modelName = modelName
    }

    override fun getVersion(): String? {
        return version
    }

    override fun getModelDir(): String {
        return modelDir!!
    }

    override fun getModelName(): String {
        return modelName!!
    }

    override fun getModelSource(): String {
        return ModelSource.LOCAL
    }

    override fun isValid(): Boolean {
        return !TextUtils.isEmpty(modelDir) &&
                !TextUtils.isEmpty(modelName)
    }

    override fun toString(): String {
        return "LocalModelOption(version=$version, modelDir=$modelDir, modelName=$modelName)"
    }


}